src/yenc.cc

```diff
@@ -78,26 +79,23 @@ PyObject* yenc_decode(PyObject* self, PyObject* Py_bytesarray_obj) {
     const char* crc_pos;
 
     // Verify it's a bytearray
-    if (!PyByteArray_Check(Py_bytesarray_obj)) {
-        PyErr_SetString(PyExc_TypeError, "Expected bytearray");
+    if (!PyMemoryView_Check(Py_memoryview_obj)) {
+        PyErr_SetString(PyExc_TypeError, "Expected memoryview");
         return NULL;
     }
 
-    if (PyObject_GetBuffer(Py_bytesarray_obj, &Py_buffer_obj, PyBUF_CONTIG) != 0) {
-        return NULL;
-    }
```

Comment:
I'm not sure about this part, by just using PyMemoryView_GET_BUFFER further down we are presuming the backing is a bytearray and contiguous.

I'd suggest PyObject_CheckBuffer and PyObject_GetBuffer - could use PyBUF_CONTIG_RO case we ever wanted memoryview(bytes(...))
Would need to add the PyBuffer_Release at the end.

Alternatively just add a PyBuffer_IsContiguous check after PyMemoryView_GET_BUFFER.

Reply:
I opted for PyBuffer_IsContiguous, a simple check.